--- a/encrypt.c	
+++ b/encrypt.c	
@@ -2,7 +2,6 @@
   This file is for Niederreiter encryption
 */
 
-#include "encrypt.h"
 
 #include "util.h"
 #include "params.h"
@@ -18,6 +17,10 @@
 #include "crypto_uint16.h"
 #include "crypto_uint32.h"
 
+/* include last because of conflict with unistd.h encrypt function */
+#include "encrypt.h"
+
+
 static inline crypto_uint16 uint16_is_smaller_declassify(uint16_t t,uint16_t u)
 {
   crypto_uint16 mask = crypto_uint16_smaller_mask(t,u);
@@ -97,7 +100,7 @@
 
 		for (j = 0; j < SYS_T; j++)
 		{
-			mask = same_mask(i, (ind[j] >> 3));
+			mask = same_mask((uint16_t)i, ind[j] >> 3);
 
 			e[i] |= val[j] & mask;
 		}
