--- avx2/sk_gen.h	2022-11-02 10:21:07.950707112 +0100
+++ fixed/sk_gen.h	2022-11-02 10:19:02.030120622 +0100
@@ -8,14 +8,12 @@
 #include "namespace.h"
 
 #define genpoly_gen CRYPTO_NAMESPACE(genpoly_gen)
-#define perm_check CRYPTO_NAMESPACE(perm_check)
 
 #include "gf.h"
 
 #include <stdint.h>
 
-int genpoly_gen(gf *, gf *);
-int perm_check(uint32_t *);
+int genpoly_gen(gf *out, gf *f);
 
 #endif
 
