--- a/benes.c	
+++ b/benes.c	
@@ -256,7 +256,7 @@
 {
 	int inc;
 
-	vec128 *b_ptr = b[0];
+	vec128 *b_ptr;
 
 	if (rev == 0) { inc =  32; b_ptr = b[ 0]; }
 	else          { inc = -32; b_ptr = b[24]; }
@@ -295,7 +295,7 @@
 	layer_3(r, b_ptr); b_ptr += inc;
 	layer_2(r, b_ptr); b_ptr += inc;
 	layer_1(r, b_ptr); b_ptr += inc;
-	layer_0(r, b_ptr); b_ptr += inc;
+	layer_0(r, b_ptr); 
 
 	transpose_64x128_sp( r );
 }
