--- a/encrypt.c	
+++ b/encrypt.c	
@@ -2,8 +2,6 @@
   This file is for Niederreiter encryption
 */
 
-#include "encrypt.h"
-
 #include "util.h"
 #include "params.h"
 #include "randombytes.h"
@@ -17,6 +15,9 @@
 #include "crypto_declassify.h"
 #include "crypto_uint32.h"
 
+/* include last because of conflict with unistd.h encrypt definition */
+#include "encrypt.h"
+
 static inline crypto_uint32 uint32_is_equal_declassify(uint32_t t,uint32_t u)
 {
   crypto_uint32 mask = crypto_uint32_equal_mask(t,u);
@@ -75,7 +76,7 @@
 
 		for (j = 0; j < SYS_T; j++)
 		{
-			mask = same_mask(i, (ind[j] >> 3));
+			mask = same_mask((uint16_t)i, ind[j] >> 3);
 
 			e[i] |= val[j] & mask;
 		}
